islandora_compound_object.module

```diff
@@ -305,48 +305,67 @@ function islandora_compound_object_update_parent_thumbnail($parent) {
 /**
  * Autocomplete callback for child object search.
  *
- * @string string $string
+ * @param string $string
  *   The user supplied string that is being searched for.
+ * @param bool $parent
+ *   A flag indicating if we are to return objects usable as parents.
  */
 function islandora_compound_object_autocomplete($string, $parent = FALSE) {
   $matches = array();
   $islandora_tuque = islandora_get_tuque_connection();
   $compound_enforcement = variable_get('islandora_compound_object_compound_children', TRUE);
 
+  // Build escapes as per:
+  // - https://www.w3.org/TR/xmlschema-2/#dt-metac and
+  // - https://www.w3.org/TR/xmlschema-2/#dt-cces1
+  $meta = array(
+    '.',
+    '\\',
```

Comment:
Ran into a weird situation where the slashes not being replaced first was causing the period to not be scaped, @adam-vessey to fix.